apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -908,6 +924,32 @@ def _add_ip_mapping_details(self, context, port, host, owned_addr,
             if host_snat_ip_allocation:
                 host_snat_ips.append(host_snat_ip_allocation)
 
+        # populate route_leak network FIPs
+        if (self._is_vrf_per_router(port_network) and
+                port_network.get(cisco_apic.ALLOW_ROUTE_LEAK, False)):
+            intf_ports = core_plugin.get_ports(
+                context,
+                filters={'device_owner':
+                         [n_constants.DEVICE_OWNER_ROUTER_INTF],
+                         'network_id': [port_network['id']]})
+            port_sbunet = self._get_port_subnets(port).pop()
+            router_ids = [x['device_id'] for x in intf_ports
+                          if x['fixed_ips'][0]['subnet_id'] == port_sbunet]
+            fixed_ips = [x['ip_address'] for x in port['fixed_ips']]
+            epg_tenant = self.apic_manager.apic.fvTenant.name(
+                self._get_tenant(port_network))
+            for router_id in router_ids:
+                fip = {'id': str(uuid.uuid4()),
```

Comment:
can we avoid a generating a new uuid during each invocation of this call, perhaps by using port id + router id? Changing uuids can cause unnecessary churn on the agent and leaf switch?